### PR TITLE
Add initial bpftool tests and another libbpf test

### PR DESCRIPTION
**Pull request by dthaler**
## Description

Add initial bpftool tests and another libbpf test

And fix various bugs hit by the tests.
This PR only adds a couple of initial tests.
A future PR will add many more tests.

bpftool_tests.exe is meant to be run on a machine with the drivers
installed.  A future PR will add it to the KM test runner in CI/CD.

## Testing

This PR adds tests.

## Documentation

None needed yet, but a future PR will add more docs on tests.

### CODE REVIEW

File: tests/bpftool_tests/bpftool_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation
```
Comment (Alan-Jowett) [inline tests/bpftool_tests/bpftool_tests.cpp:1]: Add this test to CI/CD?

**Review by Alan-Jowett — DISMISSED** [review]

File: tests/bpftool_tests/bpftool_tests.cpp (hunk 1 above)
Comment (dthaler) [reply tests/bpftool_tests/bpftool_tests.cpp:1]: The PR description already says "A future PR will add it to the KM test runner in CI/CD." :)
This is part 1 of a set.

Comment (Alan-Jowett) [reply tests/bpftool_tests/bpftool_tests.cpp:1]: OK, great. We should now be able to capture and upload code coverage from a self-hosted runner as well.

**Review by Alan-Jowett — DISMISSED** [review]

File: ebpf-for-windows.sln [hunk 2]
```diff
@@ -6,7 +6,7 @@ MinimumVisualStudioVersion = 10.0.40219.1
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "ebpfverifier", "external\ebpf-verifier\build\ebpfverifier.vcxproj", "{7D5B4E68-C0FA-3F86-9405-F6400219B440}"
 	ProjectSection(ProjectDependencies) = postProject
 		{231EE32B-EBA4-4FE5-A55B-DB18F539D403} = {231EE32B-EBA4-4FE5-A55B-DB18F539D403}
-		{98D56B8A-D8EB-3D98-B8EE-C83696B4D58A} = {98D56B8A-D8EB-3D98-B8EE-C83696B4D58A}
+		{10883FE6-9448-32DE-A899-0D98B8CA914D} = {10883FE6-9448-32DE-A899-0D98B8CA914D}
```
Comment (Alan-Jowett) [inline ebpf-for-windows.sln:9]: Why was this changed?

**Review by Alan-Jowett — APPROVED** [review]

File: ebpf-for-windows.sln (hunk 2 above)
Comment (dthaler) [reply ebpf-for-windows.sln:9]: the rebase merge did this, is it wrong?

Comment (dthaler) [reply ebpf-for-windows.sln:9]: When I look at the sln file, there is no 98D56B8A-D8EB-3D98-B8EE-C83696B4D58 project whereas 10883FE6-9448-32DE-A899-0D98B8CA914D is the yaml-cpp project which does exist, so the change looks right to me.

Comment (Alan-Jowett) [reply ebpf-for-windows.sln:9]: Is this a case where the project GUID changes depending on the build path?

Comment (dthaler) [reply ebpf-for-windows.sln:9]: > Is this a case where the project GUID changes depending on the build path?

Yes, I think so.   external\ebpf-verifier\CMakeLists.txt has:
```
# CMake derives a Visual Studio project GUID from the file path but can be overridden via a property
# (see https://gitlab.kitware.com/cmake/cmake/-/commit/c85367f4).  Using a non-constant GUID
# can cause problems if other projects/repos want to reference the ebpfverifier vcxproj file,
# so we force a constant GUID here.
set(ebpfverifier_GUID_CMAKE "7d5b4e68-c0fa-3f86-9405-f6400219b440" CACHE INTERNAL "Project GUID")
```
but there's no similar line for yaml-cpp.

File: libs/api/ebpf_api.cpp [hunk 3]
```diff
@@ -581,7 +581,10 @@ ebpf_map_get_next_key(fd_t map_fd, _In_opt_ const void* previous_key, _Out_ void
     if (result != EBPF_SUCCESS) {
         goto Exit;
     }
-    assert(key_size != 0);
+    if (key_size == 0) {
+        result = EBPF_OPERATION_NOT_SUPPORTED;
+        goto Exit;
+    }
     assert(value_size != 0);
```
Comment (saxena-anurag) [inline libs/api/ebpf_api.cpp:588]: Should we fail for this also?

**Review by saxena-anurag — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]
